### ISSUE TITLE
Compile error: parameter "num_points" was declared but never referenced

### ISSUE THREAD

**Issue by ruisebastiao** [intel-isl/Open3D#3411]
Hello, i'm getting the following error during compilation:
```
[ 66%] Building CUDA object cpp/open3d/core/CMakeFiles/core.dir/nns/FixedRadiusSearch.cu.o
/data/projects/automationstudio/external/submodules/Open3D/cpp/open3d/core/nns/FixedRadiusSearch.cu(209): error: parameter "num_points" was declared but never referenced
          detected during:
            instantiation of "void open3d::core::nns::<unnamed>::CountNeighbors(const cudaStream_t &, uint32_t *, const uint32_t *, const uint32_t *, size_t, const T *, size_t, const T *, size_t, T, T, open3d::core::nns::Metric) [with T=float]" 
(813): here
            instantiation of "void open3d::core::nns::FixedRadiusSearchCUDA(void *, size_t &, int64_t *, size_t, const T *, size_t, const T *, T, size_t, const int64_t *, size_t, const int64_t *, const uint32_t *, size_t, const uint32_t *, const uint32_t *, open3d::core::nns::NeighborSearchAllocator<T> &) [with T=float]" 
(913): here

/data/projects/automationstudio/external/submodules/Open3D/cpp/open3d/core/nns/FixedRadiusSearch.cu(354): error: parameter "num_points" was declared but never referenced
          detected during:
            instantiation of "void open3d::core::nns::<unnamed>::WriteNeighborsIndicesAndDistances(const cudaStream_t &, int32_t *, T *, const int64_t *, const uint32_t *, const uint32_t *, size_t, const T *, size_t, const T *, size_t, T, T, open3d::core::nns::Metric, __nv_bool) [with T=float]" 
(882): here
            instantiation of "void open3d::core::nns::FixedRadiusSearchCUDA(void *, size_t &, int64_t *, size_t, const T *, size_t, const T *, T, size_t, const int64_t *, size_t, const int64_t *, const uint32_t *, size_t, const uint32_t *, const uint32_t *, open3d::core::nns::NeighborSearchAllocator<T> &) [with T=float]" 
(913): here

2 errors detected in the compilation of "/data/projects/automationstudio/external/submodules/Open3D/cpp/open3d/core/nns/FixedRadiusSearch.cu".
make[2]: *** [cpp/open3d/core/CMakeFiles/core.dir/build.make:874: cpp/open3d/core/CMakeFiles/core.dir/nns/FixedRadiusSearch.cu.o] Erro 1
make[1]: *** [CMakeFiles/Makefile2:1395: cpp/open3d/core/CMakeFiles/core.dir/all] Erro 2
make: *** [Makefile:136: all] Erro 2

```
version: 0.12.0

cuda version: 11.3

**Comment by Dudulle** [comment]
This error is related to CUDA 11.3
A fix is included in #3380

**Comment by ruisebastiao** [comment]
confirmed, solved
